### PR TITLE
MongoDB Docs

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

Docs for mongodb as an online provider using cassandras docs as a template

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: docs/providers/mongodb.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Featureform supports [MongoDB](https://www.mongodb.com/) as an Inference Store.
+
+## Implementation
+
+A MongoDB table is created for every feature. A metadata table exists within the Database as well to allow the provider to keep track of its own state. Featureform's scheduler aims to achieve consistency between MongoDBs's internal state with the user's desired state as specified in the metadata service.
+
+## Configuration
+
+First we have to add a declarative MongoDB configuration in Python. In the following example, only name is required, but the other parameters are available.
```
Comment (Sami1309) [inline docs/providers/mongodb.md:11]: Wouldn't host and port also be required? Credentials? Are there combinations of valid providers?

**Review by ahmadnazeri — APPROVED** [review]

File: docs/providers/mongodb.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# MongoDB
+
+Featureform supports [MongoDB](https://www.mongodb.com/) as an Inference Store.
+
+## Implementation
+
+A MongoDB table is created for every feature. A metadata table exists within the Database as well to allow the provider to keep track of its own state. Featureform's scheduler aims to achieve consistency between MongoDBs's internal state with the user's desired state as specified in the metadata service.
```
Comment (Sami1309) [inline docs/providers/mongodb.md:7]: This paragraph is used in the other inference stores too but seems a bit wordy/convoluted. It introduces multiple elements that aren't relevant to understand the functionality. I think it is more important here to note
1. Mongodb stores values of features with their specified entity (customerID, name, timestamp) as a key
2. Featureform stores metadata on every feature registered on MongoDB, as well as its readiness for serving

**Review by Sami1309 — APPROVED** [review]
lgtm, some comments on wording

File: docs/providers/mongodb.md (hunk 1 above)
Comment (sdreyer) [reply docs/providers/mongodb.md:11]: You right

Comment (sdreyer) [reply docs/providers/mongodb.md]: Fixed

File: docs/providers/mongodb.md (hunk 2 above)
Comment (sdreyer) [reply docs/providers/mongodb.md]: Updated to be more mongo specific
